src/server/journal/journal.h

```diff
@@ -21,6 +21,10 @@ class Journal {
 
   void StartInThread();
 
+  // Starts the journal at specified LSN
+  // Also drops the (resets) the partial sync buffers
+  void StartInThreadAtLsn(LSN lsn);
```

Comment:
question: do we really need both `StartInThread` and `StartInThreadAtLsn`?  what is the case when we call StartInThread and we DO NOT want to reset partial sync buffers?

Reply:
Yes, although it's functionality can be merged (if we accept default parameters which `usually` I try to avoid).

>  what is the case when we call StartInThread

Idempotent semantics. When a replica gets registered to a master, the master tries to Start the journal. So if we reset the buffers we disabled partial sync (for each time we register a new replica)